api/v1beta1/clusterclass_types.go

```diff
@@ -465,6 +465,106 @@ type JSONSchemaProps struct {
 	// NOTE: Can be set for all types.
 	// +optional
 	Default *apiextensionsv1.JSON `json:"default,omitempty"`
+
+	// XValidations describes a list of validation rules written in the CEL expression language.
+	// +optional
+	XValidations ValidationRules `json:"x-kubernetes-validations,omitempty"`
+}
+
+// ValidationRules describes a list of validation rules written in the CEL expression language.
```

Comment:
\+1 but let's add it to the JSONSchemaProps instead because it all comes from there.

I would prefer keeping it in this file, I think we can make it clear that this comes from upstream without moving it to a separate file. Otherwise we end up with half of the CRD in a separate file which seems not necessary to me.

Reply:
Added to JSONSchemaProps.